unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarRenderer.cs

```diff
@@ -132,23 +135,38 @@ void onFailWrapper(bool isFatalError)
 
         public void InitializeImpostor()
         {
+            if (impostorInitialized)
+                return;
+
+            impostorInitialized = true;
+            impostorMesh = Instantiate(lodMeshFilter.sharedMesh);
+            impostorMaterial = new Material(lodRenderer.sharedMaterial);
+
             UserProfile userProfile = null;
+
             if (!string.IsNullOrEmpty(model?.id))
                 userProfile = UserProfileController.GetProfileByUserId(model.id);
 
             if (userProfile != null)
             {
                 bodySnapshotTexturePromise = new AssetPromise_Texture(userProfile.bodySnapshotURL);
-                bodySnapshotTexturePromise.OnSuccessEvent += asset => AvatarRendererHelpers.SetImpostorTexture(asset.texture, lodMeshFilter.mesh, lodRenderer.material);
-                bodySnapshotTexturePromise.OnFailEvent += asset => AvatarRendererHelpers.RandomizeAndApplyGenericImpostor(lodMeshFilter.mesh);
+                bodySnapshotTexturePromise.OnSuccessEvent += asset => AvatarRendererHelpers.SetImpostorTexture(asset.texture, impostorMesh, impostorMaterial);
+                bodySnapshotTexturePromise.OnFailEvent += asset => AvatarRendererHelpers.RandomizeAndApplyGenericImpostor(impostorMesh);
                 AssetPromiseKeeper_Texture.i.Keep(bodySnapshotTexturePromise);
             }
             else
             {
-                AvatarRendererHelpers.RandomizeAndApplyGenericImpostor(lodMeshFilter.mesh);
+                AvatarRendererHelpers.RandomizeAndApplyGenericImpostor(impostorMesh);
             }
         }
 
+        public void CleanImpostor()
+        {
+            impostorInitialized = false;
+            Destroy(impostorMesh);
+            Destroy(impostorMaterial);
```

Comment:
I have 0 trust in Avatar Renderer. Should we add a `null` check here?

Reply:
I didn't put a null check because `Destroy` doesn't do anything will null arguments.